src/Header.hpp

```diff
@@ -6,28 +6,46 @@
 #include <string_view>
 #include <vector>    // std::vector to store multiple header values
 
+#include "HeaderFields.hpp"
+
+/*
+	Header (RequestHeader) class holds parsed HTTP request header information.
+	
+	Contains:
+	- Request-line fields: method, path, version
+	- Header fields: stored in a HeaderFields child object
+	
+	For parsing CGI response headers (which have no request-line), 
+	use HeaderFields directly instead.
+*/
 class Header
-{ // Header class holds parsed HTTP header information
+{
 private:
-	std::string method_;						 // stores the HTTP method (e.g., GET, POST)
-	std::string path_;							 // stores the request path/URI
-	std::string version_;						 // stores the HTTP version string (e.g., HTTP/1.1)
-	std::map<std::string, std::vector<std::string>> headers_; // map of header key->list of values
+	std::string method_;	// stores the HTTP method (e.g., GET, POST)
+	std::string path_;		// stores the request path/URI
+	std::string version_;	// stores the HTTP version string (e.g., HTTP/1.1)
+	HeaderFields fields_;	// header key-value pairs
 
 public:
+	// Default constructor - creates empty header (used when parsing incrementally)
+	Header() = default;
+
 	// Construct and parse from raw HTTP request header text (request-line + header fields).
 	// Throws std::runtime_error on parse failure.
-	Header(const std::string_view &raw); // parse in constructor and throw on error
+	Header(const std::string_view &raw);
+
+	// Accessors for request-line
+	const std::string &method() const;
+	const std::string &path() const;
+	const std::string &version() const;
 
-	// Note: parsing is done in the constructor `Header(const std::string_view&)` which throws on error.
+	// Access the header fields
+	HeaderFields &fields();
+	const HeaderFields &fields() const;
```

Comment:
Since the reference provides unrestricted access, maybe `fields` should just be a public member (without underscore).